File: loja/pedidos/src/main/java/br/com/bota/entity/listener/PedidoListener.java
package br.com.bota.entity.listener;

import br.com.bota.entity.Pedido;

import javax.persistence.PrePersist;
import java.time.LocalDateTime;

public class PedidoListener {
    @PrePersist
    public void onSave(Pedido order) {
        order.setPedidoRealizadoEm(LocalDateTime.now());
    }
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/entity/listener/PedidoListener.kt
package br.com.bota.entity.listener

import br.com.bota.entity.Pedido
import java.time.LocalDateTime
import javax.persistence.PrePersist

open class PedidoListener {
    @PrePersist
    open fun onSave(pedido: Pedido) {
        pedido.pedidoRealizadoEm = LocalDateTime.now()
    }
}